squash_bot/core/response_message.py

```diff
@@ -0,0 +1,49 @@
+import enum
+import typing
+
+
+class InteractionResponseType(enum.Enum):
+    PONG = 1
+    CHANNEL_MESSAGE_WITH_SOURCE = 4
+    PREMIUM_REQUIRED = 10
+
+
+class ResponseFlags(enum.Enum):
+    """
+    Bitwise flag operations - see https://discord.com/developers/docs/resources/channel#message-object-message-flags
+    """
+
+    EPHEMERAL = 1 << 6
+    LOADING = 1 << 7
+
+
+class ResponseBody:
```

Comment:
I think we could save a decent amount of boilerplate here by using dataclasses (`attrs.frozen`) -- you get the `__init__` for free plus we could use `attrs.asdict()`

Reply:
I tried using attrs.frozen but it wouldn't work - not sure what I did wrong but maybe worth a look when next in disc.
Otherwise yeah would be good (also didn't know about `atr.asdict()` so that's cool